src/server/scripts/EasternKingdoms/ReturnToKarazhan/boss_mana_devourer.cpp

```diff
@@ -0,0 +1,318 @@
+                    events.Repeat(13700ms);
+                    break;
+                }
+                case EVENT_ENERGY_VOID:
+                {
+                    DoCastSelf(SPELL_ENERGY_VOID);
+                    events.Repeat(20600ms);
+                    break;
+                }
+                case EVENT_ENERGY_DISCHARGE:
+                {
+                    DoCastSelf(SPELL_ENERGY_DISCHARGE);
+                    events.Repeat(27900ms);
+                    break;
+                }
+                case EVENT_COALESCE_POWER:
+                {
+                    DoCastSelf(SPELL_COALESCE_POWER);
+                    events.Repeat(30300ms);
+                    break;
+                }
+                case EVENT_CHECK_POWER:
+                {
+                    if (me->GetPower(POWER_ENERGY) >= 100)
+                        DoCast(SPELL_DECIMATING_ESSENCE);
+                    events.Repeat(500ms);
+                    break;
+                }
+                default:
+                    break;
+            }
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+        }
+    }
+};
+
+// 227297 - Coalesce Power
+class spell_mana_devourer_coalesce_power : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_LOOSE_MANA_MISSILE });
+    }
+
+    void HandlePeriodic(AuraEffect const* /*aurEff*/) const
+    {
+        GetTarget()->CastSpell(GetTarget(), SPELL_LOOSE_MANA_MISSILE, TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_mana_devourer_coalesce_power::HandlePeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// 227451 - Mana Devourer Energize
+class spell_mana_devourer_energize : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MANA_DEVOURER_ENERGIZE });
+    }
+
+    void HandlePeriodic(AuraEffect const* /*aurEff*/) const
+    {
+        GetTarget()->CastSpell(GetTarget(), SPELL_MANA_DEVOURER_ENERGIZE, TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_mana_devourer_energize::HandlePeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// 227457 - Energy Discharge
+class spell_mana_devourer_energy_discharge : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spell*/) override
+    {
+        return ValidateSpellInfo({ SPELL_ENGULFING_POWER });
+    }
+
+    void HandleCast()
+    {
+        for (uint8 i = 0; i < 40; ++i)
+        {
+            float randomAngle = frand(0.0f, 2.0f * float(M_PI));
+            float randomDist = frand(GetSpellInfo()->GetMinRange(), GetSpellInfo()->GetMaxRange());
+
+            Position pos(*GetCaster());
```

Comment:
GetPosition() just returns m_position, and the Position(WorldObject const&) constructor uses the same values, but using GetPosition() is a bit more explicit and consistent with the rest of the codebase, especially when readability matters